go/libraries/doltcore/sqle/conflicts_table.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqle
+
+import (
+	"io"
+
+	"github.com/liquidata-inc/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/doltdb"
+)
+
+var _ sql.Table = (*ConflictsTable)(nil)
+
+// ConflictsTable is a sql.Table implementation that implements a system table which shows the current conflicts
+type ConflictsTable struct {
+	dbName string
+	ddb    *doltdb.DoltDB
+}
+
+// NewConflictsTable creates a ConflictsTable
+func NewConflictsTable(ctx *sql.Context, dbName string) (*ConflictsTable, error) {
+	ddb, ok := DSessFromSess(ctx.Session).GetDoltDB(dbName)
+
+	if !ok {
+		return nil, sql.ErrDatabaseNotFound.New(dbName)
+	}
+
+	return &ConflictsTable{dbName: dbName, ddb: ddb}, nil
+}
+
+// Name is a sql.Table interface function which returns the name of the table which is defined by the constant
+// ConflictsTableName
+func (dt *ConflictsTable) Name() string {
+	return doltdb.ConflictsTableName
+}
+
+// String is a sql.Table interface function which returns the name of the table which is defined by the constant
+// ConflictsTableName
+func (dt *ConflictsTable) String() string {
+	return doltdb.ConflictsTableName
+}
+
+// Schema is a sql.Table interface function that gets the sql.Schema of the log system table.
+func (dt *ConflictsTable) Schema() sql.Schema {
+	return []*sql.Column{
+		{Name: "table", Type: sql.Text, Source: doltdb.ConflictsTableName, PrimaryKey: true},
+		{Name: "num_conflicts", Type: sql.Uint64, Source: doltdb.ConflictsTableName, PrimaryKey: false},
+	}
+}
+
+type conflictsPartition struct {
```

Comment:
The original design for this was to show all the conflicts through the dolt_conflicts table... but creating the schema for that table sucked.  I may end up going back to that at some point so I didn't kill the partitions, but they may get killed in the future.

The new design will have dolt_conflicts_TABLENAME style tables in addition to this one.

Reply:
This seems fine. I don't know how else you would make a global conflicts table.